GeneratorPgn.py

```diff
@@ -0,0 +1,318 @@
+                    entry = self.replay_queue.get()
+                    self.replay_buffer.append(entry)
+                if len(self.replay_buffer) < self.replay_initial:
+                    continue
+                yield random.sample(self.replay_buffer, self.batch_size)
+        except Exception:
+            pass
+        finally:
+            self.kill_processes()
+            
+ 
+first_cords = {'a': 0, 'b': 1, 'c': 2, 'd': 3, 'e': 4, 'f': 5, 'g': 6, 'h': 7}
+second_cords = {'1': 0, '2': 1, '3': 2, '4': 3, '5': 4, '6': 5, '7': 6, '8': 7}
+pieces = {'P': 0, 'N': 1, 'B': 2, 'R': 3, 'Q': 4, 'K': 5,'p': 6, 'n': 7, 'b': 8, 'r': 9, 'q': 10, 'k': 11}
+ 
+class Preprocessor(object):
+    def __init__(self, iterator):
+        """
+        Wrapper for `Iterator` class that preprocesses compressed data to pytorch tensors.
+        :param iterator: iterator to wrap
+        """
+        super(Preprocessor, self).__init__()
+        self.iterator = iterator
+ 
+    def preprocess_fen(self, fen):
+        """
+        Preprocesses one fen and value,
+        :param fen: tuple(fen, value), data to preprocess
+        :return: dictionary of preprocessed tensors
+        """
+ 
+        raw_fen = fen
+        fen = raw_fen.split('/')
+        fen_dropped = fen[:-1]
+        fen_dropped.extend(fen[-1].split())
+        board_position_str = fen_dropped[:8]
+        next_mover_str = fen_dropped[8]
+        castling_str = fen_dropped[9]
+        en_passant_str = fen_dropped[10]
+        half_moves_str = fen_dropped[11]
+        move_idx_str = fen_dropped[12]
+ 
+        board_position = np.zeros((12, 8, 8,), dtype=np.float32)
+        for r_idx, row in enumerate(board_position_str):
+            column = 0
+            for piece in row:
+                try:
+                    p = pieces[piece]
+                    board_position[p, column, r_idx] = 1.
+                    column += 1
+                except KeyError:
+                    column += int(piece)
+ 
+        next_mover = -1 if next_mover_str == 'b' else 1
+        castling = np.zeros(4)
+        for figure in castling_str:
+            if figure == 'K':
+                castling[0] = 1
+            elif figure == 'Q':
+                castling[1] = 1
+            elif figure == 'k':
+                castling[2] = 1
+            elif figure == 'q':
+                castling[3] = 1
+        en_passant = np.zeros((1, 8, 8,), dtype=np.float32)
+        if en_passant_str != '-':
+            en_passant[0, first_cords[en_passant_str[0]], second_cords[en_passant_str[1]]] = 1.
+        half_moves = int(half_moves_str)
+        move_idx = int(move_idx_str)
+        return {'board': board_position,
+                'castling': castling,
+                'en_passant': en_passant,
+ 
+                'player': next_mover,
+                'half_move': half_moves, 'full_move': move_idx,
+                'fen': raw_fen}
+ 
+    def preprocess_batch(self, batch):

```

Comment:
(this concerns all other functions) if something should not be available to outside callers, the conventions to make it "private" in Python is to prefix the name with _